File: Implementierung/Backend/src/main/java/edu/kit/informatik/dto/userdata/rooms/RoomDto.java
package edu.kit.informatik.dto.userdata.rooms;

import edu.kit.informatik.model.userdata.rooms.Room;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import java.util.List;
import java.util.UUID;

/**
 * Data-Transfer-Object zur Klasse {@link Room}
 *
 * @author uekai
 * @author ugqbo
 * @version 1.0
 */

@ToString
@EqualsAndHashCode
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class RoomDto implements Comparable<RoomDto> {

    private String id;
    private String userId;
    private String name;
    private List<RoomObjectDto> roomObjects;

    @Override
    public int compareTo(RoomDto o) {
        UUID thisUser = UUID.fromString(this.id);
        UUID oUser = UUID.fromString(o.getId());

        return  thisUser.compareTo(oUser);
    }
}

File: Implementierung/frontend/paschi/src/dto/userdata/rooms/RoomDto.ts
import {RoomObjectDto} from "@/dto/userdata/rooms/RoomObjectDto";

export class RoomDto {

  id: string;
  userId: string;
  createdAt: string;
  updatedAt: string;
  name: string;
  roomObjects: RoomObjectDto[];
  visible: boolean;

  constructor(id: string, userId: string, createdAt: string, updatedAt: string, name: string,
              roomObjects: RoomObjectDto[], visible: boolean) {
    this.id = id;
    this.userId = userId;
    this.createdAt = createdAt;
    this.updatedAt = updatedAt;
    this.name = name;
    this.roomObjects = roomObjects;
    this.visible = visible;
  }
}
